### ISSUE TITLE
Gaussian Splatting Rendering Results

### ISSUE THREAD

**Issue by ChangyueShi** [ByteDance-Seed/Depth-Anything-3#136]
When replicating DA3's novel view rendering results, I found that the multi-view merged 3DGS are not perfectly aligned, as shown in the figure below.

<img width="1000" height="288" alt="Image" src="https://github.com/user-attachments/assets/61f3d05b-2151-4810-bd0d-4a2450634498" />

I found that the code predicts an extra offset on the point cloud positions when estimating the Gaussian's positions. After I removed this offset in the code, the point cloud alignment issue was solved, but the novel view rendering quality deteriorated, as shown in the figure below.

<img width="1018" height="288" alt="Image" src="https://github.com/user-attachments/assets/5ba18bd6-65b7-4742-9af0-535a9671ae37" />

**Comment by Jaykumaran** [comment]
Interesting, thanks for sharing this. I'm also facing issues with da3 - 3dgs.

**Comment by xjl2003** [comment]
I would greatly appreciate it if you could tell me how to load da3-giant to test the performance of 3DGS, my test code is as follows, only when model_name="da3-large" for normal operation:
import torch
from depth_anything_3.api import DepthAnything3 # , create_object, load_config
import os
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
model_name = "da3-giant"
model = DepthAnything3(model_name=model_name)
model = model.from_pretrained(os.path.join("pretrained", model_name))
model = model.to(device=device)

others model_name Error is reported: size mismatch for model.cam_enc.pose_branch.fc2.bias

**Comment by xjl2003** [comment]
It's okay, I found the problem

**Comment by ChangyueShi** [comment]
> Interesting, thanks for sharing this. I'm also facing issues with da3 - 3dgs.

Have you found a solution yet?

**Comment by donydchen** [comment]
Hi @ChangyueShi, thanks for your interest and for reporting this.

We double checked with the same DL3DV scene using the released model, and the results look reasonable on our side. Below is the script we used.

```python
import glob, os, torch
from depth_anything_3.api import DepthAnything3
device = torch.device("cuda")
model = DepthAnything3.from_pretrained("depth-anything/DA3NESTED-GIANT-LARGE")
model = model.to(device=device)
example_path = "1K/0a1b7c20a92c43c6b8954b1ac909fb2f0fa8b2997b80604bc8bbec80a1cb2da3/images_4"
images = sorted(glob.glob(os.path.join(example_path, "*.png")))[::10]  # take 31 images

output_dir = "outputs/1K_0a1b7c20"
process_res = 504
export_format = "gs_video"
export_kwargs = {"gs_video":{"trj_mode":"interpolate_smooth"}}

model.inference(
    images,
    extrinsics=None,
    intrinsics=None,
    infer_gs=True,
    export_dir=output_dir,
    process_res=process_res,
    process_res_method="upper_bound_crop",
    export_format=export_format,
    export_feat_layers=[],
    export_kwargs=export_kwargs,
)
```

When it completes, you could see `outputs/1K_0a1b7c20/gs_video/0000_interpolate_smooth.mp4`, which should match the example we provided below.

https://github.com/user-attachments/assets/c0c9baa1-9e59-4c03-afff-3298c99d1f1f

Could you kindly help run the script on your side to confirm? And feel free to let us know if you run into any other issues. Cheers.

**Comment by ChangyueShi** [comment]
> Hi [@ChangyueShi](https://github.com/ChangyueShi), thanks for your interest and for reporting this.
> 
> We double checked with the same DL3DV scene using the released model, and the results look reasonable on our side. Below is the script we used.
> 
> import glob, os, torch
> from depth_anything_3.api import DepthAnything3
> device = torch.device("cuda")
> model = DepthAnything3.from_pretrained("depth-anything/DA3NESTED-GIANT-LARGE")
> model = model.to(device=device)
> example_path = "1K/0a1b7c20a92c43c6b8954b1ac909fb2f0fa8b2997b80604bc8bbec80a1cb2da3/images_4"
> images = sorted(glob.glob(os.path.join(example_path, "*.png")))[::10]  # take 31 images
> 
> output_dir = "outputs/1K_0a1b7c20"
> process_res = 504
> export_format = "gs_video"
> export_kwargs = {"gs_video":{"trj_mode":"interpolate_smooth"}}
> 
> model.inference(
>     images,
>     extrinsics=None,
>     intrinsics=None,
>     infer_gs=True,
>     export_dir=output_dir,
>     process_res=process_res,
>     process_res_method="upper_bound_crop",
>     export_format=export_format,
>     export_feat_layers=[],
>     export_kwargs=export_kwargs,
> )
> When it completes, you could see `outputs/1K_0a1b7c20/gs_video/0000_interpolate_smooth.mp4`, which should match the example we provided below.
> 
>  0000_interpolate_smooth.mp4 
> Could you kindly help run the script on your side to confirm? And feel free to let us know if you run into any other issues. Cheers.

Thanks for the reply. I've followed your settings and it works on my side now. However, I did notice that selecting **frames 00020, 00025, 00029, 00034, and 00040 as inputs** results in geometry misalignment. Could you please verify if you observe the same issue with these inputs? Thanks again for looking into this. Below is my script: 

import glob, os, torch
from depth_anything_3.api import DepthAnything3
device = torch.device("cuda")
model = DepthAnything3.from_pretrained("depth-anything/DA3NESTED-GIANT-LARGE")
model = model.to(device=device)
example_path = "1K/0a1b7c20a92c43c6b8954b1ac909fb2f0fa8b2997b80604bc8bbec80a1cb2da3/images_4"

frame_indices = ["frame_00020", "frame_00025", "frame_00029", "frame_00034", "frame_00040"]
images = [os.path.join(example_path, f"{idx}.png") for idx in frame_indices]

output_dir = "outputs/1K_0a1b7c20"
process_res = 504
export_format = "gs_video"
export_kwargs = {"gs_video":{"trj_mode":"interpolate_smooth"}}

model.inference(
    images,
    extrinsics=None,
    intrinsics=None,
    infer_gs=True,
    export_dir=output_dir,
    process_res=process_res,
    process_res_method="upper_bound_crop",
    export_format=export_format,
    export_feat_layers=[],
    export_kwargs=export_kwargs,
)

**Comment by donydchen** [comment]
Hi @ChangyueShi, indeed, there seems to be a misalignment issue with the 5 images you mentioned. For now, you can consider switching to the updated model:

```python
model = DepthAnything3.from_pretrained("depth-anything/DA3NESTED-GIANT-LARGE-1.1")
```

This should yield better results in such a few-view scenario. I'll find time to troubleshoot what's happening with the original version.

**Comment by MinusOne2** [comment]
> Hi [@ChangyueShi](https://github.com/ChangyueShi), thanks for your interest and for reporting this.
> 
> We double checked with the same DL3DV scene using the released model, and the results look reasonable on our side. Below is the script we used.
> 
> import glob, os, torch
> from depth_anything_3.api import DepthAnything3
> device = torch.device("cuda")
> model = DepthAnything3.from_pretrained("depth-anything/DA3NESTED-GIANT-LARGE")
> model = model.to(device=device)
> example_path = "1K/0a1b7c20a92c43c6b8954b1ac909fb2f0fa8b2997b80604bc8bbec80a1cb2da3/images_4"
> images = sorted(glob.glob(os.path.join(example_path, "*.png")))[::10]  # take 31 images
> 
> output_dir = "outputs/1K_0a1b7c20"
> process_res = 504
> export_format = "gs_video"
> export_kwargs = {"gs_video":{"trj_mode":"interpolate_smooth"}}
> 
> model.inference(
>     images,
>     extrinsics=None,
>     intrinsics=None,
>     infer_gs=True,
>     export_dir=output_dir,
>     process_res=process_res,
>     process_res_method="upper_bound_crop",
>     export_format=export_format,
>     export_feat_layers=[],
>     export_kwargs=export_kwargs,
> )
> When it completes, you could see `outputs/1K_0a1b7c20/gs_video/0000_interpolate_smooth.mp4`, which should match the example we provided below.
> 
>  0000_interpolate_smooth.mp4 
> Could you kindly help run the script on your side to confirm? And feel free to let us know if you run into any other issues. Cheers.

Hi, I tried this script to infer 3D Gaussians. It works well on some scenes, but performs poorly on others.
1K/0a1b7c20a92c43c6b8954b1ac909fb2f0fa8b2997b80604bc8bbec80a1cb2da3
https://github.com/user-attachments/assets/b6d6defa-1814-4f7f-a0f9-1b90010c7e06
3K/0ad72feafa00c3a94a397dfae1d7aec5357e8f6a7db6a47c9703ff682df11e9f
https://github.com/user-attachments/assets/51bf01f3-7ed4-4dd2-893f-faa0596a69d0

**Comment by scpoppin1207** [comment]
> > Hi [@ChangyueShi](https://github.com/ChangyueShi), thanks for your interest and for reporting this.
> > We double checked with the same DL3DV scene using the released model, and the results look reasonable on our side. Below is the script we used.
> > import glob, os, torch
> > from depth_anything_3.api import DepthAnything3
> > device = torch.device("cuda")
> > model = DepthAnything3.from_pretrained("depth-anything/DA3NESTED-GIANT-LARGE")
> > model = model.to(device=device)
> > example_path = "1K/0a1b7c20a92c43c6b8954b1ac909fb2f0fa8b2997b80604bc8bbec80a1cb2da3/images_4"
> > images = sorted(glob.glob(os.path.join(example_path, "*.png")))[::10]  # take 31 images
> > output_dir = "outputs/1K_0a1b7c20"
> > process_res = 504
> > export_format = "gs_video"
> > export_kwargs = {"gs_video":{"trj_mode":"interpolate_smooth"}}
> > model.inference(
> > images,
> > extrinsics=None,
> > intrinsics=None,
> > infer_gs=True,
> > export_dir=output_dir,
> > process_res=process_res,
> > process_res_method="upper_bound_crop",
> > export_format=export_format,
> > export_feat_layers=[],
> > export_kwargs=export_kwargs,
> > )
> > When it completes, you could see `outputs/1K_0a1b7c20/gs_video/0000_interpolate_smooth.mp4`, which should match the example we provided below.
> > 0000_interpolate_smooth.mp4
> > Could you kindly help run the script on your side to confirm? And feel free to let us know if you run into any other issues. Cheers.
> 
> Hi, I tried this script to infer 3D Gaussians. It works well on some scenes, but performs poorly on others. 1K/0a1b7c20a92c43c6b8954b1ac909fb2f0fa8b2997b80604bc8bbec80a1cb2da3 https://github.com/user-attachments/assets/b6d6defa-1814-4f7f-a0f9-1b90010c7e06 3K/0ad72feafa00c3a94a397dfae1d7aec5357e8f6a7db6a47c9703ff682df11e9f https://github.com/user-attachments/assets/51bf01f3-7ed4-4dd2-893f-faa0596a69d0

I met the same issue. There are lots of ghosts and floaters in the space. And the Gaussians' opacities are quite unreasonable.
